controllers/cloud.redhat.com/providers/database/localdb.go

```diff
@@ -106,7 +106,14 @@ func (db *localDbProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error
 	}
 
 	labels := &map[string]string{"sub": "local_db"}
-	provutils.MakeLocalDB(dd, nn, app, labels, &dbCfg, image, db.Env.Spec.Providers.Database.PVC, app.Spec.Database.Name, nil)
+
+	databaseResourceRequirements := app.Spec.Database.Resources
+
+	if databaseResourceRequirements.Limits == nil && databaseResourceRequirements.Requests == nil {
+		databaseResourceRequirements = provutils.DatabaseResourceDefaults()
```

Comment:
Is there the potential for confusing behaviour here, if they only set one, their choices will get overwritten by the defaults?

Reply:
We originally had it `||`, brandon suggested `&&`. I think there are arguments for both approaches and I don't know enough about kubernetes to adjudicate between them. Logical `or` makes sense because if they provide resources but not limits what does the system use for limits? Logical `and` makes sense because principle of least surprise would require that we never stomp over user input, even if partial.

I'm happy to implement either, y'all just let me know what you think.

Reply:
I personally think you should always set both and that only setting one would be more confusing. If your app has specific resource limits, they're likely higher than usual. Setting high limits and standard requests can lead to node overcommits. The other way around can lead to high requests and low limits that might lead to OOM kills. You either set your standards or take the defaults, but that's just my take.

Reply:
I agree we could absolutely use the validating webhook here..............

Reply:
Sounded like we agreed to address webhook in a follow up MR. So, I think I'm going to revert back to `||`.